### PR TITLE
Fix: Contract data reading and parsing

### PR DESCRIPTION
**Pull request by dbrans**
## **Description**
This PR makes two fixes to reading and parsing contract data:
1. `readAddressAsContract` was incorrectly using a promise-style async call. Fixed to use callback-style async call expected by `eth-query`.
2. `determineTransactionType` was throwing `"cannot read property 'name' of undefined"` when trying to parse data. However, in the case of a simple send, data is always undefined. This PR moves the parsing to where we know we are dealing with a contract.

## **Related issues**

Tracking ticket: https://github.com/orgs/MetaMask/projects/47/views/1?pane=issue&itemId=43249800

## **Manual testing steps**

No manual testing steps.

## **Screenshots/Recordings**

Not applicable.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by dbrans** [comment]
I have read the CLA Document and I hereby sign the CLA

File: shared/modules/contract-utils.ts [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
-): Promise<contract> => {
-  let contractCode;
+): Promise<Contract> => {
+  let contractCode: string | null;
   try {
-    contractCode = await ethQuery.getCode(address);
-  } catch (e) {
+    contractCode = await pify(ethQuery.getCode.bind(ethQuery))(address);
+  } catch (err) {
+    // TODO(@dbrans): Dangerous to swallow errors here.
```
Comment (brad-decker) [inline shared/modules/contract-utils.ts:24]: @dbrans if there isn't a issue for this already its good to create one so we don't end up losing these.

File: shared/modules/transaction.utils.js [hunk 2]
```diff
@@ -139,24 +139,17 @@ export async function determineTransactionContractCode(txParams, query) {
 
 /**
  * Determines the type of the transaction by analyzing the txParams.
- * This method will return one of the types defined in shared/constants/transactions
+ * This method will return one of the types defined in {@link TransactionType}
```
Comment (brad-decker) [inline shared/modules/transaction.utils.js:142]: Noice.

**Review by brad-decker — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

File: shared/modules/contract-utils.ts (hunk 1 above)
Comment (dbrans) [reply shared/modules/contract-utils.ts:24]: Thanks @brad-decker. Created https://github.com/orgs/MetaMask/projects/47/views/1?pane=issue&itemId=43336279. Will refine.

**Pull request merged**